src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -17,18 +17,32 @@ public static class AppHost
         private const string _placeHolder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"; //hash value embedded in default apphost executable
         private readonly static byte[] _bytesToSearch = Encoding.UTF8.GetBytes(_placeHolder);
 
+        /// <summary>
+        /// Settings to customize the apphost.
+        /// </summary>
+        public class HostCustomizationSettings
+        {
+            /// <summary>
+            /// If this is set to true and the apphost is a Windows PE executable, it will have its subsystem set to GUI.
+            /// By default apphost's subsystem on Windows is set to CUI (Console).
+            /// </summary>
+            public bool WindowsGraphicalUserInterface { get; set; }
+        }
+
         /// <summary>
         /// Create an AppHost with embedded configuration of app binary location
         /// </summary>
         /// <param name="appHostSourceFilePath">The path of Apphost template, which has the place holder</param>
         /// <param name="appHostDestinationFilePath">The destination path for desired location to place, including the file name</param>
         /// <param name="appBinaryFilePath">Full path to app binary or relative path to the result apphost file</param>
         /// <param name="overwriteExisting">If override the file existed in <paramref name="appHostDestinationFilePath"/></param>
+        /// <param name="customizationSettings">Optional settings to customize the creates apphost</param>
         public static void Create(
             string appHostSourceFilePath,
             string appHostDestinationFilePath,
             string appBinaryFilePath,
-            bool overwriteExisting = false)
+            bool overwriteExisting = false,
+            HostCustomizationSettings customizationSettings = null)
```

Comment:
Continuing earlier nit, I'd rename parameter here to just `settings`. (Or  just`options` if you do take the "Options" over "Settings" in the class name.)